pkg/agent/types/annotations.go

```diff
@@ -30,6 +30,9 @@ const (
 	// ServiceExternalIPPoolAnnotationKey is the key of the Service annotation that specifies the Service's desired external IP pool.
 	ServiceExternalIPPoolAnnotationKey string = "service.antrea.io/external-ip-pool"
 
+	// ServiceAllowSharedIPAnnotationKey is the key of the Service annotation that specifies whether the Service is allowed to use shared LoadBalancer IP.
+	ServiceAllowSharedIPAnnotationKey string = "service.antrea.io/allow-shared-ip"
```

Comment:
I thought about it and didn't add it because it would be very similiar to the [MetalLB annotation](https://metallb.universe.tf/usage/#ip-address-sharing) but with a tiny difference, which may make it harder to remember if people is already familiar with the MetalLB one.

Another reason is we already use full name (load-balancer) in other annotation (load-balancer-mode), I didn't want to use different writing in this annotation while "allow-shared-load-balancer-ip" may be too long. But maybe "allow-shared-load-balancer-ip" is fine?

Reply:
changed to allow-shared-load-balancer-ip, please let me know if you prefer allow-shared-lb-ip compared with it.

Reply:
I think either one is fine (`allow-shared-load-balancer-ip` or `allow-shared-ip` to match metalLB). In the case of metalLB I would say there is a little less ambiguity because it is a dedicated LB solution, so it makes sense that "IP" would refer to LB IP in this context.
I am not really concerned about the length of the annotation (`allow-shared-load-balancer-ip`).